indexer/services/roundtable/src/tasks/create-leaderboard.ts

```diff
@@ -0,0 +1,142 @@
+import { logger } from '@dydxprotocol-indexer/base';
+import {
+  LeaderboardPnlTimeSpan,
+  LeaderboardPnlCreateObject,
+  LeaderboardPnlTable,
+  PnlTicksTable,
+  Transaction,
+} from '@dydxprotocol-indexer/postgres';
+import { LeaderboardPnlProcessedCache } from '@dydxprotocol-indexer/redis';
+import _ from 'lodash';
+
+import config from '../config';
+import { redisClient } from '../helpers/redis';
+
+export default function generateLeaderboardTaskFromTimespan(
+  timespan: LeaderboardPnlTimeSpan,
+): () => Promise<void> {
+  return async () => {
+    const leaderboardPnlProcessor: LeaderboardPnlProcessor = new LeaderboardPnlProcessor(timespan);
+    await leaderboardPnlProcessor.runTask();
+
+  };
+}
+
+class LeaderboardPnlProcessor {
+
+  constructor(
+    private timespan: LeaderboardPnlTimeSpan,
+  ) {
+    this.timespan = timespan;
+  }
+
+  async runTask(): Promise<void> {
+    await this.updateLeaderboardPnlTable(this.timespan);
+  }
+
+  /**
+   * Updates the leaderboard PnL table for a specified time span.
+   * This function performs several operations to ensure the leaderboard PnL table is up-to-date:
+   * 1. Check for updates: It first retrieves the last processed time for the given
+   *    time span from a cache. It then fetches the latest processed block time from
+   *    the PnL ticks table. If the last processed time is greater than or equal to
+   *    the latest block time, it logs a message indicating that the current PnL ticks
+   *    have already been processed for the leaderboard and exits the function.
+   *
+   * 2. Retrieve PnL Objects: If updates are needed, it retrieves
+   *    leaderboard PnL objects for the specified time span by calling
+   *    `getLeaderboardPnlObjects`.
+   *
+   * 3. Insert PnL Objects: It then inserts the leaderboard PnL objects into the
+   *    leaderboard PnL table by calling `insertLeaderboardPnlObjects`.
+   *
+   * 4. Update Cache: After successful insertion, it updates the cache with the
+   *    latest processed block time for the given time span.
+   *
+   */
+  async updateLeaderboardPnlTable(timespan: LeaderboardPnlTimeSpan) {
+    const lastProcessedTime: string | null = await LeaderboardPnlProcessedCache.getProcessedTime(
+      timespan, redisClient);
+    const lastProcessedPnlTime: string = (
+      await PnlTicksTable.findLatestProcessedBlocktimeAndCount()).maxBlockTime;
+
+    // Check if the last processed time is greater than or equal to the latest block time.
+    // In cases where indexer is from previous state, the last processed time may be null.
+    // In that case update the leaderboard.
+    if (lastProcessedTime && Date.parse(lastProcessedTime) >= Date.parse(lastProcessedPnlTime)) {
```

Comment:
is `lastProcessedTime` is null, what does this return? `Date.parse(lastProcessedTime)`

Reply:
if `lastProcessedTime` is null, it shouldnt even evaluate the second part of the && -  Date.parse(lastProcessedTime) should never be reached